libcontainer/rootfs_linux.go

```diff
@@ -107,7 +116,10 @@ func mountToRootfs(m *configs.Mount, rootfs, mountLabel string) error {
 		if err := os.MkdirAll(dest, 0755); err != nil {
 			return err
 		}
-		return syscall.Mount(m.Source, dest, m.Device, uintptr(m.Flags), "")
+		if err := syscall.Mount(m.Source, dest, m.Device, uintptr(m.Flags), ""); err != nil {
+			return err
+		}
+		return setMountPropagation(dest, syscall.MS_PRIVATE|syscall.MS_REC)
```

Comment:
Initially I had implemented something where we just let user directly specify "rshared" and then let user figure out what do they want to do. Soon I realized there are many corner cases, especially in the case of "rshared" which need to be taken care of otherwise this thing soon explodes. It took me so long to figure out various corner cases that I can't expect users to figure it out.

This is a complex feature and lot of things can go wrong. Not many people understand this feature. So we decided that instead it is better to put an abstraction layer on top and define
modes like "container_shared and container_slave". And define new behavior in terms of what to expect.  This is much more understandable. 

It is little bit higher level abstraction but atleast one can work with it. And this does not block
the possibility of exposing low level stuff directly if user needs that. Just define new values "rprivate, rslave" and let user use it. But I have travelled that road first and I myself could not figure out what to expect.